lib/matplotlib/tests/test_axes.py

```diff
@@ -434,6 +434,22 @@ def test_basic_annotate():
                 xytext=(3, 3), textcoords='offset points')
 
 
+def test_annotate_parameter_warn():
```

Comment:
You don't need all the extra plotting for testing. This should be enough:
~~~
fig, ax = plt.subplots()
with pytest.warns(MatplotlibDeprecationWarning):
    ax.annotate(s='now named text', xy=(0.5, 0.5))
~~~

Also, I recommend to use `pytest.warns(MatplotlibDeprecationWarning, match=...)` to make sure you're catching the right warning.